docs/kubernetes/ingress-controller.md

```diff
@@ -696,3 +705,81 @@ the communication work with TCP and UDP to the specified `swarm-port`:
   name: swarm-port
   protocol: TCP
 ```
+
+## Upgrades
+
+### <v0.14.0 to >=v0.14.0
+
+Kubernetes dataclient removes support for ingress v1beta1.
+What does it mean for you?
+
+1. If you run with enabled `-kubernetes-ingress-v1`, you won't need to
+  do anything and you can safely delete the flag while updating to
+  `>=0.14.0`.
+2. If you use skipper as library and pass `KubernetesIngressV1: true`
+  via `kubernetes.Options` into `kubernetes.New()`, then you won't need to
+  do anything and you can safely delete passing the option while updating to
+  `>=0.14.0`.
+3. If you use Ingress v1beta1 and run Kubernetes cluster version that
+  does not support ingress v1, then you can't update skipper to
+  `>=0.14.0`, before you upgrade your Kubernetes cluster.
+4. If you use Ingress v1beta1 and run Kubernetes cluster version that
+  support ingress v1, then you need to allow skipper to access the new
+  APIs with a changed RBAC. See the guide below.
+
+
+If you are in case 4., you have to apply a change in your RBAC, please
+check the diff or the full rendered file.
```

Comment:
I think the real migration would take multiple steps:
1. Add new rbac for ingress v1
2. Update Skipper (or enable `-kubernetes-ingress-v1`)
3. Remove v1beta1 rbac

Reply:
Ok, I see that the change adds v1 (not substitutes)

Reply:
Some people might not need it, others do, so I guess it's fine, because I assume everyone tries to think about zero downtime migrations. The fail safe is kubernetes, because one crashing pod will stop the deployment from going further.